### DISCUSSION TITLE
Configuring monitoring stack

### DISCUSSION THREAD

**Discussion by trenslow** [fluxcd/flux2#1738] [Q&A] [answered]
Hi all,

I'm working on installing the monitoring stack with the following link: https://fluxcd.io/docs/guides/monitoring/

Everything worked quite well out of the box, however I'm unsure of the best way to change some of the parameters of the installation. For instance, what if I want to disable CoreDNS scraping and enable KubeDNS scraping? What if I want to enable alertmanager? What if I want to deploy Prometheus across failover zones?

If everything were controlled by Helm, I'd simply pass different values to the kube-prometheus-stack Helm chart. But how can I configure the Helm chart when my point of control is the Kustomizations from the above link? There is even a note on that page that the Prometheus chart needs to be further configured for production, but as far as I can tell I don't see how I can configure it with this Kustomization workflow. 

Normally, I'd just do a HelmRelease workflow and configure the kube-prometheus-stack chart using the values stanza there. But in the Kustomization workflowI like how I can point to the Flux2 repo and always have the latest monitoring configurations.

Would anyone mind pointing me in the right direction? Thanks!

**Comment by trenslow** [comment, marked as answer]
Just an extra note on this. I tried to patch the kube-prometheus-stack HelmRelease using the patchesStrategicMerge field on the monitoring stack Kustomization from the tutorial. It seems to work on the first deploy of the Kustomization, but subsequent changes to the patch aren't picked up by Flux and applied. Even if I run a reconcile on the monitoring stack Kustomization, the new changes to the patch aren't applied.

**Reply by kingdonb** [reply]
I'm going to mark this as solved because the report is almost a year old, and we've fixed some issues in the Server-Side Apply behavior since 0.16-0.18 that got solved somewhere in the vicinity of Flux 0.23 that sound like your issue. I suspect this would all work if you tried it again today.

I am maintaining my own deployment of the Flux monitoring stack and I use it regularly, from the fluxcd/flux2 `manifests/monitoring` examples that are used in the [Monitoring with Prometheus guide](https://fluxcd.io/docs/guides/monitoring/) for Flux.

Please let us know if you still have any issue we can help with here!
